src/esm_parser/esm_parser.py

```diff
@@ -247,6 +248,13 @@ def initialize_from_yaml(filepath):
             user_config["general"]["runscript_abspath"] = filepath
 
             user_config = complete_config(user_config)
+    if not user_config:
+        user_error(
+            "Incorrect extension",
+            f"The runscript provided (``{filepath}``) does not have a valid file "
+            "extension. The following are valid file extensions for runscripts: "
+            f"{YAML_AUTO_EXTENSIONS[1:]}"
```

Comment:
Why the slice from 1 to the end and not just everything?

Reply:
because the first `YAML_AUTO_EXTENSION` is an empty string, and an empty string is not a valid extension. I did not know why `YAML_AUTO_EXTENSION` includes an empty string, but I did not want to dive into that right now. But maybe I should just remove the empty string from there and see what happens?

Reply:
Ah. I remember why it's empty: if the user gives a path that is a valid yaml, it doesn't need to append anything when trying to open the runscript. If not, it goes through yaml, yml, YAML, YML. 

Maybe not the best design choice...